sros2/api/__init__.py

```diff
@@ -351,13 +360,53 @@ def create_permission_file(path, name, domain_id, permissions_dict):
         </subscribe>
 """
 
+    # TODO(mikaelarguedas) remove this hardcoded handling for default parameter topics
+    service_partitions_prefix = {
+        'Request': ['', 'rq/%s' % name],
+        'Reply': ['', 'rr/%s' % name],
+    }
+    default_parameter_topics = [
+        'get_parameters',
+        'get_parameter_types',
+        'set_parameters',
+        'list_parameters',
+        'describe_parameters',
+    ]
+    for key in service_partitions_prefix.keys():
+        if key == 'Request':
+            pubsubtag = 'publish'
+        else:
+            pubsubtag = 'subscribe'
+        tag = 'partition'
+        partition_string = \
+            '<%s>' % tag + \
+            ('</%s><%s>' % (tag, tag)).join(
+                [partition for partition in service_partitions_prefix[key]]) + \
+            '</%s>' % tag
+        tag = 'topic'
+        topics_string = \
+            '<%s>' % tag + \
+            ('</%s><%s>' % (tag, tag)).join(
+                [(topic + key) for topic in default_parameter_topics]) + \
+            '</%s>' % tag
+        permission_str += """\
+        <%s>
+          <partitions>
+            %s
+          </partitions>
+          <topics>
+            %s
+          </topics>
+        </%s>
+""" % (pubsubtag, partition_string, topics_string, pubsubtag)
+
     # DCPS* is necessary for builtin data readers
     permission_str += """\
-      <subscribe>
-        <topics>
-          <topic>DCPS*</topic>
```

Comment:
I forget, what was the purpose for this rule again?
ROS2 is using the DCPS API, or for what?
https://community.rti.com/static/documentation/connext-dds/5.2.0/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/What_is_DCPS_.htm

Reply:
Connext creates DCPS topics for builtin endpoints (there's a [comment above](https://github.com/ros2/sros2/pull/33/files#diff-164a61093789ed771b5f88b17b5e4468R403)).
Note that this part is not modified in this PR, just reindented as it was improperly indented before

Reply:
I don't have that permission in my minimal working policy for talker listener demo, and it seems to messaging fine. Is there a way I can test ROS2 to fail when this permission is revoked? E.g. do I need test my use case across two different networked machines change the endpoints behavior to incoundter an issue, or are these topics optional or silently failing under the API?

Reply:
Thanks for pointing it out, my guess is that this behavior has changed in Connext 5.3.0. It was definitely necessary in 5.2.4.
As this is unrelated to this PR and will likely not be backported to ardent, let's discuss is on another ticket #34